### DISCUSSION TITLE
are people using this... successfully?

### DISCUSSION THREAD

**Discussion by colemickens** [Spotifyd/spotifyd#1280] [General]
I have spotifyd running on a number of my machines.

Spotifyd virtually *never* appears in my list of devices, unless I manually restart it.

Further, I got approximately one and a half songs into playback today before it disconnected. I gave up, I use it through Firefox most of the time anyway.

I guess I'm curious, are people actually successfully using this? With any amount of reliability?

**Comment by eladyn** [comment]
Yes, there are indeed people who are using in a sensible not-annoying way, I'm pretty sure I'm not the only one judging by the user interaction through GitHub issues or the matrix server.

However, stability is indeed a big issue, unfortunately. Especially when it comes to internet connectivity drop-outs or similar things. They aren't handled well at all at the moment and thus lead to a really bad experience for some. (Even if your connection seems stable, there might still be random disconnections that you don't notice but `spotifyd` will fail upon.)

So you basically have several options:
- see if you can somehow improve the stability of the environment that `spotifyd` is used in
- use another client
- try to improve the reconnection handling (that would be amazing, just have a look at [this discussion](https://github.com/librespot-org/librespot/discussions/609) in the upstream repository)

**Comment by colemickens** [comment]
Thank you @eladyn, that all makes sense! I'll take a look at librespot, thanks for the link to that discussion. So that I don't pollute that topic, is it accurate to guess that some of this is caused by ... Spotify itself being (overly) aggressive with respect to temporarily disconnected clients? I get the impression, from reading through that discussion, that *it* is overly sensitive to momentary blips?

**Comment by colemickens** [comment]
Well, maybe it's not that simple, since it seems that (regardless of Spotify's behavior) the librespot/spotifyd seem to often get themselves stuck to where they don't even appear as clients until a restart. 

Either way, the question is answered. Thank you for spotifyd, and thanks for the comprehensive answer. I'll close this now.
